### PR TITLE
Refactor Plan.HandleConnection to contain Plan.Handle (instead of a StorageKey)

### PR DESCRIPTION
**Pull request by mmandlis**
b/161818462

### CODE REVIEW

File: java/arcs/android/host/parcelables/ParcelablePlan.kt [hunk 1]
```diff
@@ -18,30 +18,48 @@ import arcs.core.data.Plan
 /** [Parcelable] variant of [Plan]. */
 data class ParcelablePlan(override val actual: Plan) : ActualParcelable<Plan> {
     override fun writeToParcel(parcel: Parcel, flags: Int) {
+        parcel.writeInt(actual.handles.size)
+        actual.handles.forEach {
```
Comment (piotrswigon) [inline java/arcs/android/host/parcelables/ParcelablePlan.kt:22]: You don't _need_ to write them here, as you write copies for each handle connection,  right?

File: java/arcs/android/host/parcelables/ParcelableHandle.kt [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        // TODO(161818630): read Handle's annotations.
+        return ParcelableHandle(
+            Plan.Handle(StorageKeyParser.parse(storageKeyString), type, emptyList())
+        )
+    }
+
+    override fun newArray(size: Int): Array<ParcelableHandle?> =
+        arrayOfNulls(size)
+}
+}
```
Comment (cromwellian) [inline java/arcs/android/host/parcelables/ParcelableHandle.kt:50]: weird formatting?

File: java/arcs/android/host/parcelables/ParcelablePlan.kt (hunk 1 above)
Comment (cromwellian) [reply java/arcs/android/host/parcelables/ParcelablePlan.kt:22]: True, they could be reconstructed from the HandleConnection copies I think. Don't write them, and on read, synthesize Plan.handles from mapping over HandleConnection.handle. That would solve the dup-bug

File: java/arcs/android/host/parcelables/ParcelableHandle.kt (hunk 2 above)
Comment (mmandlis) [reply java/arcs/android/host/parcelables/ParcelableHandle.kt:50]: fixed.

File: java/arcs/android/host/parcelables/ParcelablePlan.kt (hunk 1 above)
Comment (mmandlis) [reply java/arcs/android/host/parcelables/ParcelablePlan.kt]: removed.
there is still some duplication, as multiple particles may have the same handle, so keeping the bug.

**Pull request merged**